tests/test-cases/Group6-VIC-Machine/6-17-Configure-TLS.robot

```diff
@@ -0,0 +1,118 @@
+    ${output}=  Run  openssl s_client -showcerts -connect %{VCH-IP}:2378
+    Log  ${output}
+    Should Contain  ${output}  issuer=/C=US/ST=California/L=Los Angeles/O=Stark Enterprises/OU=Stark Enterprises Certificate Authority/CN=Stark Enterprises Global CA
+
+    Run  rm -rf bundle
+    Run  rm -f cert-bundle.tgz
+    Run  rm -f out-bundle
+    Run  rm -rf /root/ca
+
+
+Configure VCH - Server cert with trusted CA
+    ${domain}=  Get Environment Variable  DOMAIN  ''
+    Run Keyword If  '${domain}' == ''  Pass Execution  Skipping test - domain not set, won't generate keys
+
+    # Generate CA and wildcard cert for *.<DOMAIN>, install CA into root store
+    Generate Certificate Authority
+    Generate Wildcard Server Certificate
+    Trust Certificate Authority
+
+    ${out}=  Run  cp /root/ca/cert-bundle.tgz .; tar xvf cert-bundle.tgz
+    Log  ${out}
+
+    # Run vic-machine install, supply server cert and key
+    ${output}=  Run  bin/vic-machine-linux configure --name=%{VCH-NAME} --target="%{TEST_USERNAME}:%{TEST_PASSWORD}@%{TEST_URL}" --thumbprint=%{TEST_THUMBPRINT} --tls-key "bundle/*.%{DOMAIN}.key.pem" --tls-cert "bundle/*.%{DOMAIN}.cert.pem" ${vicmachinetls} --debug 1
+    Log  ${output}
+    Should Contain  ${output}  Loaded server certificate bundle
+    Should Contain  ${output}  Unable to locate existing CA in cert path
+    Should Contain  ${output}  Completed successfully
+
+
+    ${output}=  Run  openssl s_client -showcerts -connect %{VCH-IP}:2378
+    Log  ${output}
+    Should Contain  ${output}  issuer=/C=US/ST=California/L=Los Angeles/O=Stark Enterprises/OU=Stark Enterprises Certificate Authority/CN=Stark Enterprises Global CA
+
+    Run  rm -rf bundle
+    Run  rm -f cert-bundle.tgz
+    Run  rm -rf /root/ca
+
+    Reload Default Certificate Authorities
+
+Configure VCH - Replace certificates with self-signed certificate using --cert-path
+
+    ${domain}=  Get Environment Variable  DOMAIN  ''
+    Run Keyword If  '${domain}' == ''  Pass Execution  Skipping test - domain not set, won't generate keys
+    Run  rm -rf foo-bar-certs
+    ${output}=  Run  bin/vic-machine-linux configure --name=%{VCH-NAME} --target="%{TEST_USERNAME}:%{TEST_PASSWORD}@%{TEST_URL}" --thumbprint=%{TEST_THUMBPRINT} ${vicmachinetls} --cert-path "foo-bar-certs" --debug 1 --regenerate-certs
+
+    Should Contain  ${output}  Generating CA certificate/key pair - private key in foo-bar-certs/ca-key.pem
+    Should Contain  ${output}  Generating server certificate/key pair - private key in foo-bar-certs/server-key.pem
+    Should Contain  ${output}  Generating client certificate/key pair - private key in foo-bar-certs/key.pem
+    Should Contain  ${output}  Generated browser friendly PFX client certificate - certificate in foo-bar-certs/cert.pfx
+
+    Should Contain  ${output}  Completed successfully
+
+    ${output}=  Run  openssl s_client -showcerts -connect %{VCH-IP}:2378
+    Log  ${output}
+    Should Contain  ${output}  issuer=/O=%{VCH-NAME}
```

Comment:
Need to have cases to cover following scenarios:
- if allow user downgrade to --no-tls, verify that works
- verify if no certificate change, configure command does not mes up certificate